applications/GeoMechanicsApplication/custom_utilities/process_factory.cpp

```diff
@@ -0,0 +1,38 @@
+// KRATOS___
+//     //   ) )
+//    //         ___      ___
+//   //  ____  //___) ) //   ) )
+//  //    / / //       //   / /
+// ((____/ / ((____   ((___/ /  MECHANICS
+//
+//  License:         geo_mechanics_application/license.txt
+//
+//  Main authors:    Wijtze Pieter Kikstra
+//                   Anne van de Graaf
+//
+
+#include "custom_utilities/process_factory.hpp"
+#include "includes/kratos_parameters.h"
+
+
+namespace Kratos
+{
+
+ProcessFactory::ProductType ProcessFactory::Create(const std::string& rProcessClassName,
+                                                   const Parameters&  rProcessSettings) const
+{
+    auto pos = mCreatorMap.find(rProcessClassName);
+    if (pos == mCreatorMap.end()) {
+        return nullptr;
+    }
+
+    return pos->second ? pos->second(rProcessSettings) : nullptr;
```

Comment:
Just to verify my understanding: only here this unique_ptr is created, by calling the function that is defined when adding the creator right? I have not used functions in a map before, that's really interesting!

Reply:
Yes, only when the given process name is in the map and when the associated creator is not empty, we will create a `unique_ptr` that owns a `Process` instance. Otherwise, we get a `unique_ptr` that doesn't own anything.

What I like about this approach with the map (which I learned from a former colleague) is that it there are no changes to factory needed when you want to support other kinds of processes (even ones that don't exist yet). Just add them through `AddCreator` and you're ready to use it. A good example of the Open/Closed Principle if you'd ask me.